bin-src/lib/findChangedDependencies.ts

```diff
@@ -0,0 +1,45 @@
+import { buildDepTreeFromFiles, PkgTree } from 'snyk-nodejs-lockfile-parser';
+import { hasYarn } from 'yarn-or-npm';
+import { checkoutFile } from '../git/git';
+
+const flattenDependencyTree = (
+  tree: PkgTree['dependencies'],
+  results = new Set<string>()
+): Set<string> =>
+  Object.values(tree).reduce((acc, dep) => {
+    acc.add(`${dep.name}@@${dep.version}`);
+    return flattenDependencyTree(dep.dependencies || {}, acc);
+  }, results);
+
+// Retrieve a set of values which is in either set, but not both.
+const xor = <T>(left: Set<T>, right: Set<T>) =>
+  Array.from(right.values()).reduce((acc, value) => {
+    if (acc.has(value)) acc.delete(value);
+    else acc.add(value);
+    return acc;
+  }, new Set(left));
+
+// Yields a list of dependency names which have changed since the baseline.
+// E.g. ['react', 'react-dom', '@storybook/react']
+export const findChangedDependencies = async (baselineCommits: string[]) => {
+  const manifestName = 'package.json';
+  const lockfileName = hasYarn() ? 'yarn.lock' : 'package-lock.json';
```

Comment:
Are we going to bail in the case of `pnpm` etc?

Reply:
Yes, pnpm and Yarn pnp is out of scope.

Reply:
Sure, but we should bail out at least if we see a lockfile change for one of those.

Reply:
I think this will work out because we wouldn't be able to find any node_modules in the stats file (because they're located elsewhere), and we [already have a mechanism](https://github.com/chromaui/chromatic-cli/blob/feb1b5e87942d0bf76d6f3a4304c8b26d56422b5/bin-src/lib/getDependentStoryFiles.ts#L205-L213) to bail in case we don't find any dependency files in the stats.